traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateDataConverter.java

```diff
@@ -49,6 +82,80 @@ public HandshakeData toHandshakeData(final CertificateData certificateData) {
 		return null;
 	}
 
+	@SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.AvoidDeeplyNestedIfStmts", "PMD.NPathComplexity"})
+	public boolean verifySubject(final X509Certificate certificate, final String hostAlias ) {
+		final String host = certificate.getSubjectDN().getName();
+		if (hostCompare(hostAlias,host)) {
+			return true;
+		}
+
+		try {
+			// This approach is probably the only one that is JDK independent
+			if (certificate.getSubjectAlternativeNames() != null) {
+				for (final List<?> altName : certificate.getSubjectAlternativeNames()) {
+					if (hostCompare(hostAlias, (String) altName.get(1))) {
+						return true;
+					}
+				}
+			}
+		}
+		catch (Exception e) {
+			log.error("Encountered an error while validating the certificate subject for service: "+hostAlias+", " +
+					"error: "+e.getClass().getSimpleName()+": " + e.getMessage(), e);
+			return false;
+		}
+
+		return false;
+	}
+
+	private boolean hostCompare(final String hostAlias, final String subject) {
+		if (hostAlias.contains(subject) || subject.contains(hostAlias)) {
+			return true;
+		}
+		final String[] chopped = subject.split("CN=", 2);
+		String chop = null;
+		if (chopped != null && chopped.length > 1) {
+			chop = chopped[1];
+		}
+		if (chop != null) {
```

Comment:
Can't the body of this `if` be inside the body of the previous `if` (i.e. right under `chop = chopped[1];`)? `chop` can only not be `null` if that first `if`'s condition is `true`. For that matter, you could also just declare chop inside of that as `String chop = chopped[1];` since it's not going to be used outside of the `if` block.

Reply:
You are correct. It turns out that the split actually returns an empty string if there is no token. So in the new commit I have also added and extra check to make sure 'chop' has length.